plugin/storage/grpc/shared/grpc_client.go

```diff
@@ -34,6 +35,18 @@ type grpcClient struct {
 	depsReaderClient storage_v1.DependenciesReaderPluginClient
 }
 
+// updateContextWithBearerToken updates the outgoing context with a bearer token extracted from the source
+func (c *grpcClient) updateContextWithBearerToken(outgoing context.Context, source context.Context) context.Context {
+	bearerToken, hasToken := spanstore.GetBearerToken(source)
+	if hasToken {
+		requestMetadata := metadata.New(map[string]string{
+			"bearerToken": bearerToken,
```

Comment:
A suggestion here. There is a place in the code where the property identifying this setting is already defined. It is: https://github.com/jaegertracing/jaeger/blob/master/storage/spanstore/token_propagation.go#L21.
Maybe we could do something like:

```golang
const BearerTokenKey = "bearer.token"
const bearerToken = contextKey("bearer.token")
```

and use the `spanstore.BearerTokenKey` here and then in the gRPC plugin server?

Reply:
+1

Reply:
Will do. Final question: should I squash before it is merged?

Reply:
No need, we squash on merge